apps/src/blockly/addons/cdoFieldDropdown.js

```diff
@@ -32,6 +32,38 @@ export default class CdoFieldDropdown extends GoogleBlockly.FieldDropdown {
     }
   }
 
+  /**
+   * Sets the field's state based on the given state value. Should only be
+   * called by the serialization system.
+   *
+   * @param state The state we want to apply to the field.
+   * @override because different labs store `state` in different ways.
+   * For music lab, `state` is the value of the field.
+   * For other labs, `state` is stringified xml.
```

Comment:
Also, I believe it's fine for state to be stringified xml i this case and that it's actually an intentional part of mainline's backwards compatibility with XML..

Reply:
Thanks Mike! Music lab is doing what the Blockly team now recommends (using json) and Blocky's implementation of the fields is to support both json and xml. But they didn't anticipate this scenario where sometimes state is xml and sometimes it's stored in json as it is in our code base.
I think this was a super useful bug since we're migrating to using json, but from my understanding, we'll always have to support xml to some degree (levelbuilder, student projects).